collect_app/src/main/java/org/odk/collect/android/widgets/StringWidget.java

```diff
@@ -104,6 +108,22 @@ protected StringWidget(Context context, FormEntryPrompt prompt, boolean readOnly
         }
 
         addAnswerView(answerText);
+        answerText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                widgetValueChanged();
```

Comment:
Shouldn't we call this method in clearAnswer() as well?

Reply:
`setText` results in `afterTextChanged` being called. If we call `widgetValueChanged()` in `clearAnswer` as well, it will get called twice.